File: tests/expectations/body.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

typedef enum {
  Foo1,
  Bar1,
  Baz1,
} MyCLikeEnum;

typedef enum {
  Foo1_Prepended,
  Bar1_Prepended,
  Baz1_Prepended,
} MyCLikeEnum_Prepended;

typedef struct {
  int32_t i;
#ifdef __cplusplus
  inline void foo();
#endif
} MyFancyStruct;

typedef enum {
  Foo,
  Bar,
  Baz,
} MyFancyEnum_Tag;

typedef struct {
  int32_t _0;
} Bar_Body;

typedef struct {
  int32_t _0;
} Baz_Body;

typedef struct {
  MyFancyEnum_Tag tag;
  union {
    Bar_Body bar;
    Baz_Body baz;
  };
#ifdef __cplusplus
  inline void wohoo();
#endif
} MyFancyEnum;

typedef union {
  float f;
  uint32_t u;
  int32_t extra_member; // yolo
} MyUnion;

typedef struct {
#ifdef __cplusplus
  inline void prepended_wohoo();
#endif
  int32_t i;
} MyFancyStruct_Prepended;

typedef enum {
  Foo_Prepended,
  Bar_Prepended,
  Baz_Prepended,
} MyFancyEnum_Prepended_Tag;

typedef struct {
  int32_t _0;
} Bar_Prepended_Body;

typedef struct {
  int32_t _0;
} Baz_Prepended_Body;

typedef struct {
  #ifdef __cplusplus
    inline void wohoo();
  #endif
  MyFancyEnum_Prepended_Tag tag;
  union {
    Bar_Prepended_Body bar_prepended;
    Baz_Prepended_Body baz_prepended;
  };
} MyFancyEnum_Prepended;

typedef union {
  int32_t extra_member; // yolo
  float f;
  uint32_t u;
} MyUnion_Prepended;

void root(MyFancyStruct s,
          MyFancyEnum e,
          MyCLikeEnum c,
          MyUnion u,
          MyFancyStruct_Prepended sp,
          MyFancyEnum_Prepended ep,
          MyCLikeEnum_Prepended cp,
          MyUnion_Prepended up);

File: tests/expectations/body.zig
const std = @import("std");

pub const MyCLikeEnum = extern union(c_uint) {
    Foo1,
    Bar1,
    Baz1,
};

pub const MyCLikeEnum_Prepended = extern union(c_uint) {
    Foo1_Prepended,
    Bar1_Prepended,
    Baz1_Prepended,
};

pub const MyFancyStruct = extern struct {
    i: i32,
};

pub const MyFancyEnum_Tag = extern union(c_uint) {
    Foo,
    Bar,
    Baz,
};

const struct_unnamed_2 = extern struct {
    bar: i32,
};
const struct_unnamed_3 = extern struct {
    baz: i32,
};
const union_unnamed_1 = extern union {
    unnamed_0: struct_unnamed_2,
    unnamed_1: struct_unnamed_3,
};
pub const MyFancyEnum = extern struct {
    tag: MyFancyEnum_Tag,
    unnamed_0: union_unnamed_1,
};
pub const MyUnion = extern union {
    f: f32,
    u: u32,
    extra_member: i32,
};
pub const MyFancyStruct_Prepended = extern struct {
    i: i32,
};

pub const MyFancyEnum_Prepended_Tag = extern union(c_uint) {
    Foo_Prepended,
    Bar_Prepended,
    Baz_Prepended,
};

const struct_unnamed_5 = extern struct {
    bar_prepended: i32,
};
const struct_unnamed_6 = extern struct {
    baz_prepended: i32,
};
const union_unnamed_4 = extern union {
    unnamed_0: struct_unnamed_5,
    unnamed_1: struct_unnamed_6,
};
pub const MyFancyEnum_Prepended = extern struct {
    tag: MyFancyEnum_Prepended_Tag,
    unnamed_0: union_unnamed_4,
};

pub const MyUnion_Prepended = extern union {
    extra_member: i32,
    f: f32,
    u: u32,
};

extern fn root(s: MyFancyStruct, e: MyFancyEnum, c: MyCLikeEnum, u: MyUnion, sp: MyFancyStruct_Prepended, ep: MyFancyEnum_Prepended, cp: MyCLikeEnum_Prepended, up: MyUnion_Prepended) anyopaque;
